File: script/game.js
define(["jquery", "lib/gl-matrix", "resourceManager/ResourceManager", "sceneGraph/Scene", "sceneGraph/ObjectNode", "sceneGraph/TransformNode"],
    function ($, glm, ResourceManager, Scene, ObjectNode, TransformNode) {
        "use strict";

        var Game = function () {
            this.canvas = $("#canvas")[0];
            this.canvas.width = window.innerWidth;
            this.canvas.height = window.innerHeight;

            this.gl = this.canvas.getContext("experimental-webgl", {antialias: true});
            this.gl.aniso = this.gl.getExtension('EXT_texture_filter_anisotropic');
//            this.gl.fl = this.gl.getExtension("OES_texture_float");
//            this.gl.hfl = this.gl.getExtension("OES_texture_half_float");

            this.resourceManager = new ResourceManager(this.gl);
            this.scene = new Scene(this.resourceManager, this.gl, this.canvas.width, this.canvas.height);

            var thi$ = this;
            var model = this.resourceManager.getResource({url: "models/krejt.json"}, ResourceManager.ResourceType.MODEL,
                function (oModel) {
                    var move = new TransformNode(function (mMatrix) {
                        return glm.mat4.translate(mMatrix, mMatrix, [0, 0, 0]);
                    });
                    thi$.scene.root.append(move);

                    var rotate = new TransformNode(function (mMatrix) {
                        return glm.mat4.rotate(mMatrix, mMatrix, Math.PI / 180 * rotate.r, [1, 2, 3]);
                    });
                    rotate.r = 0;
                    move.append(rotate);

                    var object = new ObjectNode(oModel);
                    rotate.append(object);

                    thi$.scene.camera.moveTo([0, 0, 7]);

                    thi$.scene.refresh();
                }
            );
        };

        Game.prototype.frame = function () {

            this.scene.render();
        };

        return Game;
    }
);

File: lib/game.dart
library game;

import 'dart:html';
import 'dart:web_gl' as gl;

import 'package:vector_math/vector_math.dart';

import 'package:Twer/resource_management/resource_management.dart';
import 'package:Twer/scene_graph/scene_graph.dart';

class Game {
  CanvasElement canvas;

  gl.RenderingContext context;

  ResourceManager resourceManager;
  Scene scene;

  double rotation = 0.0;

  Game() {
    this.canvas = document.getElementById('canvas');
    this.canvas.width = window.innerWidth;
    this.canvas.height = window.innerHeight;

    this.context = canvas.getContext3d();

    this.resourceManager = new ResourceManager(this.context);
    this.scene = new Scene(this.resourceManager, this.context,
    this.canvas.width, this.canvas.height);

    Model model = this.resourceManager.getResource(ResourceManager.MODEL,
    resourceURL:'models/krejt.json');

    TransformNode move = new TransformNode((
        Matrix4 matrix) => matrix.translate(0.0, 0.0, 0.0));
    this.scene.root.append(move);

    TransformNode rotate = new TransformNode((
        Matrix4 matrix) => matrix.rotate(new Vector3(1.0, 2.0, 3.0),
    this.rotation));
    move.append(rotate);

    rotate.append(new ObjectNode(model));

    this.scene.camera.moveTo(new Vector3(0.0, 0.0, 7.0));

    this.scene.refresh();
  }
}